src/libcommon/log/customlogwstream.h

```diff
@@ -26,19 +26,43 @@ class CustomLogWStream : private std::wstringstream {
         CustomLogWStream &operator=(const CustomLogWStream &) = delete;
 
         std::wstring str() const { return std::basic_stringstream<wchar_t>::str(); }
+
+        template<class T>
+        CustomLogWStream &operator<<(T) = delete;
+
         CustomLogWStream &operator<<(const wchar_t *str) {
             // We need to cast to std::wstringstream as the operator<<(std::wstringstream, const wchar_t *str) is defined outside
             // of the class std::wstringstream and therefore it is not applicable to the current object because of the private
             // inheritance
             static_cast<std::wstringstream &>(*this) << str;
             return *this;
         }
+        CustomLogWStream &operator<<(bool b) {
+            std::wstringstream::operator<<(b);
+            return *this;
+        }
         CustomLogWStream &operator<<(int i) {
             std::wstringstream::operator<<(i);
             return *this;
         }
+        CustomLogWStream &operator<<(int64_t i64) {
+            std::wstringstream::operator<<(i64);
+            return *this;
+        }
+        CustomLogWStream &operator<<(size_t si) {
+            std::wstringstream::operator<<(si);
+            return *this;
+        }
+        CustomLogWStream &operator<<(unsigned long ul) {
+            std::wstringstream::operator<<(ul);
+            return *this;
+        }
+        CustomLogWStream &operator<<(double d) {
+            std::wstringstream::operator<<(d);
+            return *this;
+        }
         CustomLogWStream &operator<<(const std::wstring &str) {
-            std::wstringstream::operator<<(str.c_str());
+            CustomLogWStream::operator<<(str.c_str());
```

Comment:
Is it useful or required to call `c_str()`?

Reply:
As for CustomLogWStream &operator<<(const wchar_t *str), the operator is defined outside of std::wstringstream. Here is a commit where I used the same workaround for both CustomLogWStream &operator<<(const wchar_t *str) and CustomLogWStream &operator<<(const std::wstring &str).

Reply:
[893c0da](https://github.com/Infomaniak/desktop-kDrive/pull/321/commits/893c0da73b65eade834cfb5c0fb4eba7b42ab79f)